### ISSUE TITLE
Failed to build webcam_face_pose_ex

### ISSUE THREAD

**Issue by babylon73** [davisking/dlib#2031]
OpenCV (4.2.0) is installed via these steps (on Windows):
1. Got it from sourceforge.net
2. Unpacked from self-extracting executable
3. Extended PATH system variable with "......\opencv\opencv\build\x64\vc15\bin" path
4. Created OPENCV_DIR user variable = "......\opencv\opencv\build\x64\vc15\lib"

## Expected Behavior
webcam_face_pose_ex builds successfully

## Current Behavior
Getting these errors:
Building Custom Rule dlib/dlib/examples/CMakeLists.txt
webcam_face_pose_ex.cpp
dlib\dlib\dlib\opencv/cv_image.h(37): error C2440: 'initializing': cannot convert from 'const cv::Mat' to 'IplImage' [dlib\dlib\examples\build\webcam_face_pose_ex.vcxproj]
dlib\dlib\dlib\opencv/cv_image.h(37): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
dlib\dlib\dlib\opencv/cv_image.h(26): note: while compiling class template member function 'dlib::cv_image<dlib::bgr_pixel>::cv_image(const cv::Mat)'
dlib\dlib\examples\webcam_face_pose_ex.cpp(73): note: see reference to function template instantiation 'dlib::cv_image<dlib::bgr_pixel>::cv_image(const cv::Mat)' being compiled
dlib\dlib\examples\webcam_face_pose_ex.cpp(73): note: see reference to class template instantiation 'dlib::cv_image<dlib::bgr_pixel>' being compiled

## Steps to Reproduce
Just build the full batch of examples

* **Version**: 19.19
* **Where did you get dlib**: dlib.net
* **Platform**: Win10 x64
* **Compiler**: VS2017

**Comment by davisking** [comment]
Try the latest dlib on github.

**Comment by babylon73** [comment]
Indeed, it is compilable now, thank you.

I have connected an IP camera via rtsp protocol and I'm facing this: an example starts just fine and it works for some time (but very slow, may be 10 times slower than real life camera speed), however, it stops working with h264 errors after may be 15-20 seconds of action:
[h264 @ 0000013b3c3f4580] error while decoding MB 73 42, bytestream -12
[h264 @ 0000013b3c3f4e80] error while decoding MB 46 15, bytestream -19

May be it's because the stream is handled very slowly, I'm not sure. All the examples are compiled with AVX instructions enabled, CUDA is also enabled:
cmake -G "Visual Studio 15 2017 Win64" -T host=x64 .. -DDLIB_USE_CUDA=1 -DUSE_AVX_INSTRUCTIONS=1
-- Using CMake version: 3.16.2
-- Compiling dlib version: 19.19.99
-- Enabling AVX instructions
-- Looking for cuDNN install...
-- Found cuDNN: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.2/lib/x64/cudnn.lib
-- Enabling CUDA support for dlib.  DLIB WILL USE CUDA

I have tried to comment the face detection code in order to forward the vide stream into the window without any processing - and now it works without slowdowns and h264 does not fail.

May be you could suggest something... seems like it's just the slow processing that kills the example.

Hardware is pretty sufficient though: Threadripper 2950X + 2080 Ti.

Thank you!

**Comment by davisking** [comment]
Maybe http://dlib.net/faq.html#Whyisdlibslow

**Comment by babylon73** [comment]
I have tripple-checked before asking...

That's how I'm building:
mkdir build
cd build
cmake -G "Visual Studio 15 2017 Win64" -T host=x64 .. -DDLIB_USE_CUDA=1 -DUSE_AVX_INSTRUCTIONS=1
cmake --build . --config Release

I have also tried to build with -DUSE_SSE4_INSTRUCTIONS=1, same speed, same result.

I have tried to build from Visual Studio (using examples.sln generated by CMake), made sure it's release - same speed.

This is the line that takes remarkable time to operate:
std::vector<rectangle> faces = detector(cimg);

As the stream from the camera has time stamp embedded, I can see that the time in frames changes 4-5 times slower than real time.

No slowdowns when face detection is commented.
